docker/hail-ubuntu/Dockerfile

```diff
@@ -16,10 +16,18 @@ RUN chmod 755 /bin/retry && \
     hail-apt-get-install curl gpg && \
     curl 'https://keyserver.ubuntu.com/pks/lookup?search=0xF23C5A6CF475977595C89F51BA6932366A755776&hash=on&exact=on&options=mr&op=get' \
          | gpg --dearmor > /usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg && \
+    curl 'https://keyserver.ubuntu.com/pks/lookup?search=0xF23C5A6CF475977595C89F51BA6932366A755776&hash=on&exact=on&options=mr&op=get' \
+         | gpg --dearmor > /usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg && \
+    curl 'https://apt.llvm.org/llvm-snapshot.gpg.key' \
+        | gpg --dearmor >> /usr/share/keyrings/llvm-snapshot-keyring.gpg && \
     echo 'deb [signed-by=/usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg] http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal main' \
          >> /etc/apt/sources.list && \
     echo 'deb-src [signed-by=/usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg] http://ppa.launchpad.net/deadsnakes/ppa/ubuntu focal main' \
          >> /etc/apt/sources.list && \
+    echo 'deb [signed-by=/usr/share/keyrings/llvm-snapshot-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main' \
+         >> /etc/apt/sources.list && \
+    echo 'deb-src [signed-by=/usr/share/keyrings/llvm-snapshot-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main' \
+         >> /etc/apt/sources.list && \
```

Comment:
Are these working? I thought the implicit behavior of ubuntu in docker nowadays is to clear the sources list after install to reduce layer size. Either way, I think it'd make more sense to keep these in the same `RUN` directive as what installs the llvm toolchain.

Reply:
These were working. The build breaks because we use pip to install PyYAML 6 and llvm-15-dev depends on python3-yaml.

Reply:
Could we solve that by not building the `query-build` image on base_image and instead base it on hail_ubuntu? Should be lighter weight just will probably need to replicate a few apt installs

Reply:
good point. I'll do that.